### PR TITLE
add new static camera mode

### PR DESCRIPTION
**Pull request by lianghongzhuo**
add new static camera mode

### CODE REVIEW

File: src/camera_positioner.cpp [hunk 1]
```diff
@@ -5,114 +5,145 @@
...
+                {
+                    if (std::find(bundle_tags.begin(), bundle_tags.end(), msg.detections[i].id[0]) != bundle_tags.end())
+                    {
+                        tf::Transform bundle_transform;
+                        tf::poseMsgToTF(msg.detections[i].pose.pose.pose, bundle_transform);
+                        if (initialized=="start")
+                        {
+                            ROS_INFO("camera positioner is running");
+                            if(get_tf_only_on_start)
+                                    initialized = "fixed_mode";
```
Comment (v4hn) [inline src/camera_positioner.cpp]: The name `initialized` does not make sense anymore if you store the operation mode here.
Should probably be `state` or `mode` instead.

File: src/camera_positioner.cpp [hunk 2]
```diff
@@ -5,114 +5,145 @@
...
+                                else
+                                    initialized = "moraml_mode";
```
Comment (v4hn) [inline src/camera_positioner.cpp]: "moraml"?

File: src/camera_positioner.cpp [hunk 3]
```diff
@@ -5,114 +5,145 @@
...
+
+        if(ros::Time::now() - latest_detection_time > ros::Duration(20.0))
+        {
+            ROS_WARN_THROTTLE(5, "Didn't detect apriltag bundle for camera position update in 20 seconds. The camera might have moved in the meanwhile.");
+        }
+
+        // if we measured the camera's position successfully, publish it
+        if(initialized != "start")
+        {
+            br.sendTransform(tf::StampedTransform(world_camera_transform, ros::Time::now(), "/world", camera_link));
```
Comment (v4hn) [inline src/camera_positioner.cpp:141]: For your new "fixed" mode, publishing the transform in the callback means you have to receive apriltag detection messages *constantly* in order to publish your known values (and ignore the messages).
That does not make sense at all.

In your mode, you should instead shutdown the subscriber altogether (also to show you do not use the values)
and instead spawn an `std::thread` that publishes the transform at a fixed frequency as long as `ros::ok()`.

**Review by v4hn — CHANGES_REQUESTED** [review]
Your patch changes about every single line of formatting for no obvious reason.
Usually, you should keep the formatting style of the existing code instead!

The commits are also not clean, in the sense that, e.g., the first commit that "changes formatting" also introduces a new `arg` to the launch files.

Lastly, the logic does not make much sense (see inline comment).

The code was never very nice, but we really should try not to make it worse. :)

**Comment by lianghongzhuo** [comment]
@v4hn Thank you for your review, I will try to update the code.

**Comment by lianghongzhuo** [comment]
@v4hn I updated the code as your comment except for the code indent. 
As the current code has a mixed code indent (2 and 3 spaces). So I make these indent consistently with 4 spaces.

**Comment by lianghongzhuo** [comment]
after call sub.shutdow() tf is not publishing anymore, even if I write a thread, strange. Any suggestions?

**Pull request closed without merging**